### PR TITLE
Fix BC break when baseControllerName uses bundle notation with subfolder

### PR DESCRIPTION
**Pull request by tifabien**
I am targeting this branch, because it fixes a BC break when using Bundle notation with Subfolder.

## Changelog

```markdown
### Fixed
- Fixed BC break when baseControllerName uses bundle notation with subfolder
```
## Subject

Related to #5228

### CODE REVIEW

File: src/Route/RouteCollection.php [hunk 1]
```diff
@@ -84,6 +84,10 @@ public function add(
 
         if (!isset($defaults['_controller'])) {
             $actionJoiner = false === \strpos($this->baseControllerName, '\\') ? ':' : '::';
+            if (false !== \strpos($this->baseControllerName, '\\') && false !== \strpos($this->baseControllerName, ':')) {
```
Comment (kunicmarko20) [inline src/Route/RouteCollection.php]: I don't like having 2 same checks `false !== \strpos($this->baseControllerName, '\\')`

Comment (tifabien) [reply src/Route/RouteCollection.php]: What is your suggestion?

Comment (kunicmarko20) [reply src/Route/RouteCollection.php]: Maybe do:

```
if ($actionJoiner !== ':' && ...) {
```

**Comment by greg0ire** [comment]
Thanks @tifabien !

**Pull request merged**